sweyer_plugin/lib/sweyer_plugin_platform_interface.dart

```diff
@@ -5,6 +5,44 @@ import 'package:plugin_platform_interface/plugin_platform_interface.dart';
 
 import 'sweyer_plugin_method_channel.dart';
 
+
+/// Base class for all exceptions from the Sweyer plugin.
+class SweyerPluginException implements Exception {
+  /// The exception reason.
+  final String message;
+  /// An optional cause of this exception.
+  final Exception? cause;
```

Comment:
Why is it optional?

Reply:
We might want to throw other exceptions in the future without having a causing exception.